eth/filters/filter.go

```diff
@@ -266,6 +267,9 @@ func (f *Filter) checkMatches(ctx context.Context, header *types.Header) (logs [
 	if err != nil {
 		return nil, err
 	}
+	if logsList == nil {
```

Comment:
It's weird to add this "duplicated" check here. I would vote to add the check  here https://github.com/ethereum/go-ethereum/pull/25459/files#diff-9886da3412b43831145f62cec6e895eb3613a175b945e5b026543b7463454603R204

Also in `les_apiBackend.GetLogs`, we should also return the error if logs is not found.

```go
func (b *LesApiBackend) GetLogs(ctx context.Context, hash common.Hash) ([][]*types.Log, error) {
	if number := rawdb.ReadHeaderNumber(b.eth.chainDb, hash); number != nil {
		return light.GetBlockLogs(ctx, b.eth.odr, hash, *number)
	}
	return nil, fmt.Errorf("failed to get logs for block #%d", header.Number.Uint64())
}
```